src/js/events/MesosSummaryActions.js

```diff
@@ -104,12 +104,17 @@ var MesosSummaryActions = {
     function (resolve, reject) {
 
       return function (timeScale) {
-        let canAccessHistoryServer = Hooks.applyFilter(
+        let canAccessHistoryAPI = Hooks.applyFilter(
           'hasCapability', false, 'historyServiceAPI'
         );
 
-        if (!_historyServiceOnline || !canAccessHistoryServer) {
-          requestFromMesos(resolve, reject);
+        if (!_historyServiceOnline || !canAccessHistoryAPI) {
+          let canAccessMesosAPI = Hooks.applyFilter(
+            'hasCapability', false, 'mesosAPI'
+          );
+          if (canAccessMesosAPI) {
```

Comment:
Yup, the latter would be preferable. Hmm... not sure what a good temporary solution would look like.
Probably emit another event that is then handled in the MesosSummaryStore separately.
But that would probably be the same as what you are doing here.

How urgent is this? Could we make sure to make this change as we migrate to Relay? Or would it be more beneficial to have the actual fix in place before we start the migration?